backend/src/sockets/handler.ts

```diff
@@ -1,8 +1,13 @@
 import { Server, Socket } from "socket.io";
 import { Event } from "./event";
+import axios from "axios";
+import logger from "../config/logger";
+import qs from "querystring";
+import Video from "../models/video";
+import uniqid from "uniqid";
 
 type EventHandler = {
-  [event in Event]?: Function;
+  [event in Event]?: (io: Server, socket: Socket, roomId: string, args: any) => Promise<void>;
```

Comment:
Changed this to a promise so that the events could trigger `async` functions like HTTP requests.